hnn_core/lfp.py

```diff
@@ -0,0 +1,234 @@
+Amrita University, India.
+
+Email: harilalp@am.amrita.edu; shyam@amrita.edu
+www.amrita.edu/compneuro
+
+translated to Python and modified to use use_fast_imem by Sam Neymotin
+based on mhines code
+
+References
+----------
+1 . Parasuram H, Nair B, D'Angelo E, Hines M, Naldi G, Diwakar S (2016)
+Computational Modeling of Single Neuron Extracellular Electric Potentials
+and Network Local Field Potentials using LFPsim. Front Comput Neurosci 10:65
+[PubMed].
+"""
+
+# Authors: Mainak Jas <mjas@mgh.harvard.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Sam Neymotin <samnemo@gmail.com>
+
+from neuron import h
+
+import numpy as np
+from numpy.linalg import norm
+
+
+def _get_sections_on_this_rank(sec_type='Pyr'):
+    ls = h.allsec()
+    ls = [s for s in ls if sec_type in s.name()]
+    return ls
+
+
+def _get_segment_counts(all_sections):
+    '''The segment count of a section excludes the endpoints (0, 1)'''
+    seg_counts = list()
+    for sec in all_sections:
+        seg_counts.append(sec.nseg)
+    return seg_counts
+
+
+def _transfer_resistance(section, ele_pos, sigma, method):
+    """Transfer resistance between section and electrode position.
+
+    To arrive at the extracellular potential, the value returned by this
+    function is multiplied by the net transmembrane current flowing through all
+    segments of the section. Hence the term "resistance" (voltage equals
+    current times resistance).
+
+    Parameters
+    ----------
+    section : h.Section() The NEURON section. ele_pos : list (x, y, z) The x,
+        y, z coordinates of the electrode (in um) sigma : float Extracellular
+        conductivity (in S/m) method : str Approximation to use. 'psa' assigns
+        all transmembrane currents to the center point (0.5). 'lsa' treats the
+        section as a line source, but a single multiplier is calculated for
+        each section. Returns
+    -------
+    vres : list The resistance at each section.
+    """
+    ele_pos = np.array(ele_pos)  # electrode position to Numpy
+
+    start = np.array([section.x3d(0), section.y3d(0), section.z3d(0)])
+    end = np.array([section.x3d(1), section.y3d(1), section.z3d(1)])
+
+    if method == 'psa':
+
+        mid = (start + end) / 2.
+
+        # distance from section midpoint to electrode
+        dis = norm(ele_pos - mid)
+
+        # setting radius limit
+        if dis < section.diam / 2.0:
+            dis = section.diam / 2.0 + 0.1
+
+        phi = 1. / dis
+
+    elif method == 'lsa':
+        # From: Appendix C (pp. 137) in Holt, G. R. A critical reexamination of
+        # some assumptions and implications of cable theory in neurobiology.
+        # CalTech, PhD Thesis (1998).
+        #
+        #                      Electrode position
+        #   |------ L --------*
+        #                 b / | R
+        #                 /   |
+        #   0==== a ====1- H -+
+        #
+        # a: section orientation vector
```

Comment:
```suggestion
        # a: section position vector
```

no? see: https://en.wikipedia.org/wiki/Position_(geometry)

Reply:
The "orientation vector" is defined by P('1') - P('0'), where P('x') are the "position vectors" of the two points defining `a`. The vector P('1') - P('0') is the position vector of point P('1') iff. P('0') is the origin.

Reply:
Wow sorry @cjayb I was trying to undo something and deleted your comment! I'll summarize here, please correct as needed:

Since `a` is calculated as `P(1) - P(0)` it would only be considered the position vector if `P(0)` is considered the origin.

Reply:
In any case, orientation seems to imply the topologic concept of orientability, and the orientation vector specifically refers to an axis of rotation https://en.wikipedia.org/wiki/Orientation_(geometry)#Orientation_vector

Reply:
Okay I see what you mean. I would say then:

"Vector oriented along the section"

Reply:
deal :)